lib/Platform/Intl/LocaleMatcher.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+@interface LocaleMatcher : NSObject
+
++ (NSString *)bestAvailableLocale:(NSString *)locale:(NSArray<NSString *> *)availableLocales;
++ (NSString *)getBestAvailableLocale:(std::vector<std::u16string>)requestedLocales;
```

Comment:
I think as far as possible, we should implement this in C++ and use C++ `u16string`s. Especially for utility functions like this, there is a good chance we can use a C++ implementation to also replace parts of our Java implementation, which would reduce code duplication across the platforms.